src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.NodeMapBuilder.cs

```diff
@@ -255,6 +255,19 @@ public override BoundNode VisitBinaryOperator(BoundBinaryOperator node)
                 throw ExceptionUtilities.Unreachable;
             }
 
+            public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
```

Comment:
The node map allows mapping from syntax back to bound node. 
From discussion with Neal, what we want in the map in the case of a deconstruction declaration is: a node for the whole assignment expression, nodes for the bound types that appear on the LHS, and a node for the RHS value. Note that no nodes are added to the map for the locals being declared.
This is similar to what happens for a local declaration (`var x = 1;`), which also does not put anything for `x` in the map.

But the bound nodes for deconstruction hold more information than is typical, such as deconstruction steps, conversion and assignment steps. Those don't map directly to syntax though, and so should not appear in the map for the semantic model to understand the syntax.